modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/DurationThresholdConfig.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * -\-\-
+ * Spotify Apollo Metrics Module
+ * --
+ * Copyright (C) 2013 - 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo.metrics.semantic;
+
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigObject;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DurationThresholdConfig {
+
+  private final Optional<Integer> allEndpoints;
+  private final Map<String, Map<String,
+      Integer>> goals;
+
+  private DurationThresholdConfig(final Optional<Integer> allEndpoints,
+                                  final Map<String, Map<String,
+                                      Integer>> goals) {
+    this.allEndpoints = allEndpoints;
+    this.goals = goals;
+  }
+
+  public Optional<Integer> getDurationThresholdForEndpoint(final String method,
+                                                           final String endpoint) {
+    final Optional<Integer> endpointGoal =
+        Optional.ofNullable(goals.getOrDefault(endpoint, ImmutableMap.of()).getOrDefault
+            (method, null));
+    if (endpointGoal.isPresent()) {
+      return endpointGoal;
+    } else if (allEndpoints.isPresent()) {
+      return allEndpoints;
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  public Optional<Integer> getDurationThresholdForEndpoint(final String endpointMethod) {
+    final String[] endpointMethodSplit = endpointMethod.split(":");
+    return getDurationThresholdForEndpoint(endpointMethodSplit[0], endpointMethodSplit[1]);
+  }
+
+  public static DurationThresholdConfig parseConfig(final Config config) {
+    final Optional<Integer> allEndpoints;
+    final Map<String, Map<String, Integer>> goals;
+    if (config.hasPath("endpoint-duration-goal")) {
+      final ConfigObject configObject = config.getObject("endpoint-duration-goal");
```

Comment:
I believe you can write this as:
```
      config.getObject("endpoint-duration-goal")
          .withoutKey("all-endpoints")
          .unwrapped()
          .entrySet()
          .stream()
          .collect(
              Collectors.toMap(
                  entry -> doSomething(entry.getKey()),
                  entry -> doSomething((Map<String, Integer>) entry.getValue())));
```

which is also where you could flatten the map.